### PR TITLE
IC-1312 get eosr by referral

### PR DESCRIPTION
**Pull request by rgforsyth**
## What does this pull request do?

Adds an endpoint to get an end of service report by referral id

## What is the intent behind these changes?

To be able to retrieve an end of service report for a referral.

### CODE REVIEW

**Comment by sldblog** [comment]
I'm a bit confused, does this overlap with #143?

**Comment by rgforsyth** [comment]
> I'm a bit confused, does this overlap with #143?

This story is to retrieve an end of service report by a referral id. The other story is to retrieve a eosr by the id of an eosr.
Each of these stories has been branched off the previous one. As they haven't been merged yet, the work from each is included in each pr if that's what you mean?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/EndOfServiceReportController.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.UUID
+
+@RestController
+class EndOfServiceReportController(
+  val jwtAuthUserMapper: JwtAuthUserMapper,
+  val locationMapper: LocationMapper,
+  val endOfServiceReportService: EndOfServiceReportService,
+) {
+
+  @PostMapping("/referral/{id}/end-of-service-report")
```
Comment (sldblog) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/EndOfServiceReportController.kt:23]: Given our intent here is to "start an end of service report for a referral", that relate to similar intents like "start an action plan for a referral", which we implemented as `POST /draft-action-plan`

Should we use that pattern here, too, to be consistent?

**Comment by sldblog** [comment]
> As they haven't been merged yet, the work from each is included in each pr if that's what you mean?

Yes, thanks for clarifying. Using the other PR's branch as the base branch for this PR might be useful in these situations, if that's helpful for explicit ordering?

> wants to merge 1 commit into `IC-1309-get-eosr` from `IC-1312-get-eosr-by-referral-id`

**Pull request merged**